File: ios/CaboGame/Models/Player.swift
import Foundation

struct Player: Identifiable, Codable, Equatable {
    let id: UUID
    var name: String
    var hand: [Card?]
    var roundsToSkip: Int

    init(id: UUID = UUID(), name: String, hand: [Card?] = [], roundsToSkip: Int = 0) {
        self.id = id
        self.name = name
        self.hand = hand
        self.roundsToSkip = roundsToSkip
    }

    func score() -> Int {
        hand.compactMap { $0 }.reduce(0) { $0 + $1.rank.caboValue }
    }
}

File: android/app/src/main/kotlin/com/navitech/cabo/models/Player.kt
package com.navitech.cabo.models

import kotlinx.serialization.Serializable
import java.util.UUID

@Serializable
data class Player(
    val id: String = UUID.randomUUID().toString(),
    val name: String,
    val hand: List<Card?> = emptyList(),
    val roundsToSkip: Int = 0
) {
    fun score(): Int = hand.filterNotNull().sumOf { it.rank.caboValue }
}
